### ISSUE TITLE
Support for updated gnucash SQL schema

### ISSUE THREAD

**Issue by atx** [sdementen/piecash#126]
I have GnuCash 3.7 and opening the sqlite file fails with `ValueError: Unsupported table versions`. This seems to be caused by an update in the database schema. See below for details:

```
sqlite> SELECT * FROM versions ORDER BY table_name;
Gnucash|3000001
Gnucash-Resave|19920
accounts|1
billterms|2
books|1
budget_amounts|1
budgets|1
commodities|1
customers|2
employees|2
entries|4
invoices|4
jobs|1
lots|2
orders|1
prices|3
recurrences|2
schedxactions|1
slots|4
splits|5
taxtable_entries|3
taxtables|2
transactions|4
vendors|1
```

**Comment by hiromu2000** [comment]
I also updated to 3.7 and started having "ValueError: Unsupported table versions".

**Comment by gorf** [comment]
I got the same error. And temporary fixed it by changing the variable   version_supported.
version_supported = {
    '2.6': {
        'Gnucash': 2062100,
        'Gnucash-Resave': 19920,
        'accounts': 1,
        'billterms': 2,
        'books': 1,
        'budget_amounts': 1,
        'budgets': 1,
        'commodities': 1,
        'customers': 2,
        'employees': 2,
        'entries': 3,
        'invoices': 3,
        'jobs': 1,
        'lots': 2,
        'orders': 1,
        'prices': 2,
        'recurrences': 2,
        'schedxactions': 1,
        'slots': 3,
        'splits': 4,
        'taxtable_entries': 3,
        'taxtables': 2,
        'transactions': 3,
        'vendors': 1,
    },
    '3.0': {
        'Gnucash': 3000001,
        'Gnucash-Resave': 19920,
        'accounts': 1,
        'billterms': 2,
        'books': 1,
        'budget_amounts': 1,
        'budgets': 1,
        'commodities': 1,
        'customers': 2,
        'employees': 2,
        'entries': 4,
        'invoices': 4,
        'jobs': 1,
        'lots': 2,
        'orders': 1,
        'prices': 3,
        'recurrences': 2,
        'schedxactions': 1,
        'slots': 4,
        'splits': 5,
        'taxtable_entries': 3,
        'taxtables': 2,
        'transactions': 4,
        'vendors': 1,
    }
}

**Comment by Zburatorul** [comment]
@gorf, thanks!

**Comment by sdementen** [comment]
Could you try the branch Py3/gnucash3 to see if it fixes the issue?

**Comment by OliverEvans96** [comment]
@sdementen - the py3/gnucash3 branch does not work for me, but @gorf's fix works. The diff from master is 

```
diff --git a/piecash/core/session.py b/piecash/core/session.py
index 0fa321b..371454f 100644
--- a/piecash/core/session.py
+++ b/piecash/core/session.py
@@ -48,7 +48,7 @@ version_supported = {
         'vendors': 1,
     },
     '3.0': {
-        'Gnucash': 3000000,
+        'Gnucash': 3000001,
         'Gnucash-Resave': 19920,
         'accounts': 1,
         'billterms': 2,
@@ -67,7 +67,7 @@ version_supported = {
         'recurrences': 2,
         'schedxactions': 1,
         'slots': 4,
-        'splits': 4,
+        'splits': 5,
         'taxtable_entries': 3,
         'taxtables': 2,
         'transactions': 4,
```

**Comment by sdementen** [comment]
Indeed, the table version of splits has been increased to 5 in commit https://github.com/Gnucash/gnucash/commit/e88f31e05cd57ba6fcc07563d98fb55c5310d25d

**Comment by gregwalters** [comment]
Closing this issue, it was merged as part of #130 and continues on in #136 for changes to pip.
